File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n=nums.length;

        int i=0;
        int j=0;
        int idx=0;

        while(j<n){
            if(nums[i]!=nums[j]){
                if(j-i==1){
                    nums[idx]=nums[i];
                    i=j;
                    idx++;
                }
                else if(j-i>=2){
                    nums[idx]=nums[i];
                    idx++;
                    nums[idx]=nums[i];
                    i=j;
                    idx++;
                }
            }
            j++;
        }
        if(j-i==1){
            nums[idx]=nums[i];
            i=j;
            idx++;
        }
        else if(j-i>=2){
            nums[idx]=nums[i];
            idx++;
            nums[idx]=nums[i];
            i=j;
            idx++;
        }

        return idx;
    }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n=nums.size();

        int i=0;
        int j=0;
        int k=0;

        while(j<n){
            while(j<n && nums[i]==nums[j]) j++;
            int len=j-i;
            if(len>2) len=2;

            while(len>0){
                nums[k]=nums[i];
                len--;
                k++;
            }

            i=j;
        }
        return k;
    }
};